### PR TITLE
epics-base: Edit build configuration files to use the Spack compiler wrappers

### PR DESCRIPTION
**Pull request by eflumerf**

### CODE REVIEW

**Comment by eflumerf** [comment]
@glenn-horton-smith I discovered that epics-base was using /usr/bin/gcc and /usr/bin/g++ for its Spack build, even when I requested %gcc@11.3.0. This also results in the RPATHs not being set for the EPICS libraries, which may have been causing issues with downstream code linking against libCA

**Comment by glenn-horton-smith** [comment]
Thanks Eric!

**Review by glenn-horton-smith — APPROVED** [review]
Looks good. Thanks!

**Review by glenn-horton-smith — APPROVED** [review]
OK with me.

**Review by glenn-horton-smith — APPROVED** [review]
Looks good.

**Review by glenn-horton-smith — APPROVED** [review]
Still looks good.

**Review by glenn-horton-smith — APPROVED** [review]
I am approving the changes, but github continues to say "Review required" and "You’re not [authorized](https://docs.github.com/articles/about-protected-branches/) to merge this pull request."  I've read the article linked in the later message and still don't see exactly what the issue is.

**Comment by glenn-horton-smith** [comment]
Apparently I don't have write access, so my approval is not enough. Could someone with write access to [spack:develop](https://github.com/spack/spack/tree/develop) please approve this PR?

File: var/spack/repos/builtin/packages/epics-base/package.py
Comment (scheibelp) [inline var/spack/repos/builtin/packages/epics-base/package.py]: Note that `self.compiler.cxx` is the actual compiler binary *underneath* the Spack compiler wrapper. If you want this to be the spack wrapper you can set `"CC = " + spack_cc` and `... = spack_cxx`

Comment (scheibelp) [reply var/spack/repos/builtin/packages/epics-base/package.py]: (these variables are automatically defined for you and globally available to all methods in a package file)

Comment (eflumerf) [reply var/spack/repos/builtin/packages/epics-base/package.py]: Thanks for the tip! I saw both in https://spack.readthedocs.io/en/latest/packaging_guide.html#filtering-functions so was a little confused.

**Comment by scheibelp** [comment]
@spackbot run pipeline

**Review by scheibelp — APPROVED** [review]

**Comment by glenn-horton-smith** [comment]
Thanks Peter and Eric.

**Comment by eflumerf** [comment]
@scheibelp Can this be merged?

**Review by scheibelp — APPROVED** [review]

**Comment by scheibelp** [comment]
Thanks for the ping (and apologies for losing track of this).

**Pull request merged**
